src/algokit/cli/generate.py

```diff
@@ -10,6 +11,7 @@
 logger = logging.getLogger(__name__)
 
 
+@cache
```

Comment:
How long does this cache for? I assume just the current invocation?

Reply:
Yes just the current invocation. its a simplified version of lru_cache from the same functools, i believe this was added in 3.9 initially. 

The difference from lru_cache is that there is no size limit on the cache, the decorator provides an in-memory cache (a dictionary) that stores the results of the function calls using the arguments of the calls as the keys and the results as the values. If the function is called again with the same arguments, functools.cache retrieves the result from the cache instead of executing the function again.

That would not have been ideal if we were to use it outside of cli context but in our case each call to cli invalidates cache. Given that we override click group class, i added this to remove duplicated call to the same methods because currently they are invoked both from get_command and list_commands